### PR TITLE
All latest changes

### PR DESCRIPTION
**Pull request by schernolyas**
Attempt N3

### CODE REVIEW

File: orientdb/pom.xml [hunk 1]
```diff
@@ -63,21 +63,25 @@
 
         <dependency>
             <groupId>com.orientechnologies</groupId>
-            <artifactId>orientdb-core</artifactId>
+            <artifactId>orientdb-core</artifactId>
```
Comment (DavideD) [inline orientdb/pom.xml]: This kind of chges should not be included in a PR. I don't think you need to change this file BTW.
Is there something you need to add to this POM to work with OrientDB?

File: orientdb/pom.xml [hunk 2]
```diff
@@ -63,21 +63,25 @@
...
+        <dependency>
+            <groupId>org.parboiled</groupId>
+            <artifactId>parboiled-java</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.hibernate</groupId>
-            <artifactId>hibernate-search-orm</artifactId>
-            <scope>test</scope>
+            <artifactId>hibernate-search-engine</artifactId>
+            <type>jar</type>
```
Comment (DavideD) [inline orientdb/pom.xml]: Why?

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 3]
```diff
@@ -19,9 +21,16 @@
 import org.hibernate.datastore.ogm.orientdb.impl.OrientDBDatastoreProvider;
 import org.hibernate.datastore.ogm.orientdb.logging.impl.Log;
 import org.hibernate.datastore.ogm.orientdb.logging.impl.LoggerFactory;
+import org.hibernate.datastore.ogm.orientdb.query.impl.OrientDBParameterMetadataBuilder;
 import org.hibernate.datastore.ogm.orientdb.type.spi.ORecordIdGridType;
 import org.hibernate.engine.spi.SessionFactoryImplementor;
 import org.hibernate.ogm.dialect.identity.spi.IdentityColumnAwareGridDialect;
+import org.hibernate.ogm.dialect.multiget.spi.MultigetGridDialect;
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: Is this import needed? We decided to post-pone the use of `MultigetGridDialect`

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 4]
```diff
@@ -63,25 +92,13 @@ public OrientDBDialect(OrientDBDatastoreProvider provider) {
...
-				dbKey = (ORecordId) columnValue;
-			}
-		}
-
-		if ( dbKey.getClusterPosition() == ORecordId.EMPTY_RECORD_ID.getClusterPosition() ) {
-			// is is temporary value. no entity in db with this key
-			log.info( "getTuple:Key:" + dbKey + "is temporary!" );
-			return null;
-		}
+		log.info( "getTuple:EntityKey:" + key + "; tupleContext" + tupleContext + " tupleContext.getClass():" + tupleContext.getClass() );
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: This log should not be at info level

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 5]
```diff
@@ -92,94 +109,285 @@ public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
...
+
+/*	@Override
+	public List<Tuple> getTuples(EntityKey[] keys, TupleContext tupleContext) {
+		ArrayList<Tuple> tuples = new ArrayList<>( keys.length );
+		for ( EntityKey key : keys ) {
+			log.info( "getTuples:EntityKey:" + key + "; tupleContext" + tupleContext );
+			tuples.add( getTuple( key, tupleContext ) );
+		}
+		return tuples;
+	} */
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: Can you remove this comments and add this code only when we decide to use it?

File: orientdb/pom.xml (hunk 1 above)
Comment (schernolyas) [inline orientdb/pom.xml]: I have fixed add checkstyle remarks in next PR

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java [hunk 6]
```diff
@@ -63,25 +92,13 @@ public OrientDBDialect(OrientDBDatastoreProvider provider) {
...
-			// is is temporary value. no entity in db with this key
-			log.info( "getTuple:Key:" + dbKey + "is temporary!" );
-			return null;
-		}
+		log.info( "getTuple:EntityKey:" + key + "; tupleContext" + tupleContext + " tupleContext.getClass():" + tupleContext.getClass() );
 
 		try {
-			Map<String, Object> dbValuesMap = entityQueries.get( key.getMetadata() ).findEntity( provider.getConnection(), key.getColumnValues() );
+                        Map<String, Object> dbValuesMap = entityQueries.get( key.getMetadata() ).findEntity( provider.getConnection(), key );
+			if ( dbValuesMap == null || ( dbValuesMap != null && dbValuesMap.isEmpty() ) ) {
```
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: This should be:
```
if ( dbValuesMap == null || dbValuesMap.isEmpty()  ) { 
```

File: orientdb/pom.xml (hunk 2 above)
Comment (schernolyas) [inline orientdb/pom.xml]: I added hibernate-search-engine and hibernate-search-orm  besause  some tests with using JPQL queris,  uses classes from these artifacts

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 3 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: No. imports not clean.

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 4 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: ofcourse!  All logs will reorganized. Must of them is for analyse Hibernate working

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 6 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: Yes.  You right

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 4 above)
Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: You can change the log level in the properties file in the test, you don't have to add debug code as info

File: orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java (hunk 5 above)
Comment (schernolyas) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: in theory, I can. But ...

Okey  as You wish. I remove it

Comment (DavideD) [inline orientdb/src/main/java/org/hibernate/datastore/ogm/orientdb/OrientDBDialect.java]: Note that you can alway keep this in a separate branch and include them when the time comes

**Pull request closed without merging**